content/post/partiels-fonction-hugo.md

```diff
@@ -0,0 +1,346 @@
+2. La fonction `range` ne bronchera pas avec un tableau vide, mais tout autre type de valeur entraînerait une erreur de génération. Il est donc toujours plus sage de tester à l'aide d'un `with`, sauf si vous êtes vraiment sûrs de retourner un tableau.
+3. Nous appelons notre précédent partiel de fonction, mais cette fois nous le mettons en cache. Pour les variantes, nous utilisons les deux valeurs de ses arguments.
+4. Nous sauvegardons le tout dans un tableau associatif à des fins de lisibilité. Puisque notre `$tag` est déclaré dans le contexte de notre boucle `range`, il ne pourra pas entrer en conflit avec un autre `$tag` comme par exemple le prochain tag de la liste.
+5. Nous utilisons la fonction [`append`](https://gohugo.io/functions/append/#readout) pour ajouter notre tableau associatif `$tag` au tableau que nous retournons.
+6. 🎉
+
+Maintenant dans notre modèle nous pouvons écrire :
+
+```go-html-template
+# layouts/_default/single.html
+{{/* 1. */}}
+{{ range partial "func/GetTags" $ }}
+  {{/* 2. */}}
+  <a href="{{ .URL }}">{{ .Name }}</a>
+{{ end }}
+```
+
+1. Nous avons que la valeur retournée par notre partiel de fonction maison est à coup sûr un tableau, vide ou non. Nous pouvons donc utiliser `range` sans problème.
+2. Nous pouvons maintenant utiliser les clefs personnalisées de nos tableaux associatifs.
+3. C'est tout !
+
+## Des améliorations possibles ?
+
+Bien entendu ! Nous pourrions :
+
+- Exclure certains tags du tableau retourné par la fonction `GetTags`
+- Transformer `GetTags` en `GetTerms`, afin de pouvoir l'utiliser pour n'importe quelle taxonomie.
+- Trouver la bonne variante de notre *partiel de fonction* `GetTags` et utiliser `partialCached`.
+- Développer bien plus de partiels de fonction pour répondre à d'autres besoins !
+
+## Conclusion
+
+Après avoir vu les bases, nous avons pu développer deux partiels de fonction qui nous aideront grandement dans la maintenance de l'affichage des taxonomies de notre site.
+
+Et si nous avons besoin d’afficher seulement certains articles ou bien tous les articles mais en excluant certains tags ? Cela se passera dans la fonction `GetTags` et pas ailleurs !  Et si dans une prochaine version Hugo introduit un moyen plus efficace de gérer les termes d’une taxonomie ? Nous ajusteront notre fonction `GetTerm` !
+
+Avec ses *partiels de fonction*, Hugo répond enfin à la séparation des problématiques de templating et de gestion des données, en permettant la réutilisabilité et le typage de données !
+
+Est-ce que je vous ai déjà dit que c'était une de mes fonctionnalités préférées dont je vais abuser en 2020 ?
+
+Si vous avez un retour d’expérience ou des questions à propos des partiels de fonction, ou que voulez simplement partager les partiels que vous avez développé suite à lecture de cet article, [laissez un commentaire ou un bout de code](https://regisphilibert.com/blog/2019/12/hugo-partial-series-part-2-functions-with-returning-partials/) !
```

Comment:
C'est pas facile de suivre si on n'a jamais idée de comment sont stockées les données manipulées, il me semble que de monter à quoi ressemble un tableau associatif de taxonomies ou autre structure plus complexe serait utile à la compréhension des fonctions manipulées (ou via l'ajout de quelques `printf` pour faciliter le suivi)

Reply:
Le partiel s'appelle toujours `GetTags` alors qu'on peut affichier les catégories, pourquoi ne pas le renommer `GetTaxonomies` ?

Reply:
Non elle s'appele `GetTerm` après. (Parce que c'est terme que l'ont "get" pas une taxonomie.

Reply:
Ah attends, oui après pardon... Je m'embrouille. Ben j'ai voulu faire simple 🤷‍♂ mais c'est peut-être plus perturbant qu'autre chose.

Reply:
Je précise en suite dans les chose à améliorer: _Turn GetTags into GetTerms, making it taxonomy agnostic._

Reply:
Au temps pour moi :)

Reply:
`categories` et `tags` sont des taxonomies non ?

Reply:
Oui mais "Art" "Litterature" sont des "terms" en tout cas dans le jargon Hugo.